test/sanity/validate-modules/fix_isinstance.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python
+
+# A helper script that patches the simple usage of "type(foo) == bar" in modules.
+
+import os
+import re
+import shutil
+import sys
+import subprocess
+import tempfile
+
+TYPE_REGEX = re.compile(r'.*\stype\(.*')
+r1 = re.compile(r'.*\stype\(.*\)\s+is\s+(int|bool|str|dict|list|tuple|MethodType|vim\..*):')
+r2 = re.compile(r'.*\stype\(.*\)\s+is\snot\s+(int|bool|str|dict|list|tuple|MethodType|vim\..*):')
+r3 = re.compile(r'.*\stype\(.*\)\s+==\s+(int|bool|str|dict|list|tuple|MethodType|vim\..*):')
+r4 = re.compile(r'.*\stype\(.*\)\s+!=\s+(int|bool|str|dict|list|tuple|MethodType|vim\..*):')
+r5 = re.compile(r'.*\stype\(.*\)\s+is\s+not\s+(int|bool|str|dict|list|tuple|MethodType|vim\..*):')
+
+def run_command(args):
+    p = subprocess.Popen(
+            args,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            shell=True
+        )
+    (so, se) = p.communicate()
+    return (p.returncode, so, se)
+
+def find_modules(basepath):
+    cmd = 'find %s -type f -name "*.py"' % basepath
+    (rc, so, se) = run_command(cmd)
+    files = [x.strip() for x in so.split('\n') if x.strip()]
+    files = [x for x in files if not x.endswith('__init__.py')]
+    files = [x for x in files if not '.git' in x]
+    files = [x for x in files if not '/test/' in x]
```

Comment:
Just a neat python thing that you might be unaware of.  Python has generator expressions which are a quick way to create and chain iterators.  They can act similar to shell pipelines if you don't need actual lists but can just act on iterators the whole way through.  It saves on memory usage sometimes.  Rewritten as generator expressions, this becomes:
``` python
+    files = (x.strip() for x in so.split('\n') if x.strip())
+    files = (x for x in files if not x.endswith('__init__.py'))
+    files = (x for x in files if not '.git' in x)
+    files = [x for x in files if not '/test/' in x]
```
The last one remains a list comprehension so that you can return a list rather than an iterator.
(This is just informational, not a blocker.)
```